src/image/image.js

```diff
@@ -143,82 +143,35 @@ p5.prototype.createImage = function(width, height) {
  *  @param  {String} [filename]
  *  @param  {String} [extension]
  */
-p5.prototype.saveCanvas = function() {
-  var cnv, filename, extension;
-  if (arguments.length === 3) {
-    cnv = arguments[0];
-    filename = arguments[1];
-    extension = arguments[2];
-  } else if (arguments.length === 2) {
-    if (typeof arguments[0] === 'object') {
-      cnv = arguments[0];
-      filename = arguments[1];
-    } else {
-      filename = arguments[0];
-      extension = arguments[1];
-    }
-  } else if (arguments.length === 1) {
-    if (typeof arguments[0] === 'object') {
-      cnv = arguments[0];
-    } else {
-      filename = arguments[0];
-    }
-  }
-
+p5.prototype.saveCanvas = function(cnv, filename, extension) {
   if (cnv instanceof p5.Element) {
     cnv = cnv.elt;
-  }
-  if (!(cnv instanceof HTMLCanvasElement)) {
-    cnv = null;
-  }
-
-  if (!extension) {
-    extension = p5.prototype._checkFileExtension(filename, extension)[1];
-    if (extension === '') {
-      extension = 'png';
-    }
+  } else if (!(cnv instanceof HTMLCanvasElement)) {
+    filename = cnv;
+    extension = filename;
+    cnv = this._curElement && this._curElement.elt;
   }
 
-  if (!cnv) {
-    if (this._curElement && this._curElement.elt) {
-      cnv = this._curElement.elt;
-    }
-  }
+  extension =
+    extension ||
+    p5.prototype._checkFileExtension(filename, extension)[1] ||
+    'png';
 
-  if (p5.prototype._isSafari()) {
-    var aText = 'Hello, Safari user!\n';
-    aText += 'Now capturing a screenshot...\n';
-    aText += 'To save this image,\n';
-    aText += 'go to File --> Save As.\n';
-    alert(aText);
-    window.location.href = cnv.toDataURL();
-  } else {
-    var mimeType;
-    if (typeof extension === 'undefined') {
-      extension = 'png';
+  var mimeType;
+  switch (extension) {
+    default:
```

Comment:
Shouldn't `default` case be at the bottom of the `switch` block? Otherwise it will always trip on `default` and `break` here, making mime type always `image/png`. Or you can just remove the `break` statement as well if that feels more efficient.

Reply:
I'm not sure prcisely what you're saying, but i think you might be confused about how switch statements work: https://codepen.io/Spongman/pen/WXmMaP

Reply:
`default` can be anywhere in the switch, but it's usually placed at the end because of fallthrough and it being easier to think of it like taking the first match. Move it to the end to reduce cognitive load when reading the code 😃

Reply:
Ah yes, @Zalastax is right, I'm not used to reading switch statement like that. Sorry

Reply:
yeah, the eslint recommended rules `no-fallthrough` and `no-duplicate-case` ensure that case fallthrough and ordering should no longer be an issue.